source/connection_manager.c

```diff
@@ -801,9 +794,69 @@ static void s_schedule_connection_culling(struct aws_http_connection_manager *ma
     }
     aws_mutex_unlock(&manager->lock);
 
-    aws_event_loop_schedule_task_future(manager->cull_event_loop, manager->cull_task, cull_task_time);
+    return cull_task_time;
+}
+
+static uint64_t s_calculate_pending_connections_acquire_cull_task_time(struct aws_http_connection_manager *manager) {
+    if (manager->pending_connections_acquire_timeout_ms == 0) {
+        return 0;
+    }
+
+    uint64_t cull_task_time = 0;
+    aws_mutex_lock(&manager->lock);
+
+    const struct aws_linked_list_node *end = aws_linked_list_end(&manager->pending_acquisitions);
+    struct aws_linked_list_node *oldest_node = aws_linked_list_begin(&manager->pending_acquisitions);
+    if (oldest_node != end) {
+        /*
+         * Since the pending acquires are in FIFO order in the list, the front of the list has the closest
+         * cull time.
+         */
```

Comment:
it's just confusing that the comments above is
```
        /*
         * Since the connections are in LIFO order in the list, the front of the list has the closest
         * cull time.
         */
```
And here, everything is the same but FIFO order to be mentioned...

I think for the acquisition it's not as complicated as the connections. It won't back to the queue after poped. So, it may be simpler to just mention that the front will be the closest cull time.

Reply:
Thanks, updated with the simple version.